### PR TITLE
Allow reorg with special api key

### PR DESCRIPTION
**Pull request by AshtonStephens**
## Description

Closes: #1036 

## Changes

Adds a new env variable to the AWS lambda to pick up the api key that is allowed to perform reorgs. This will make it such that only a specific trusted api key can cause a reorg to mess up the chainstate.

This was done by adding a new parameter to all the functions that can initiate a reorg to detect the api key and check it against the approved api key.

## Testing Information

This was tested with a deployed version of the emily api in a non-production account. It was first tested with a disallowed api key on the chainstate integration tests and ONLY the reorg tests failed (great) and then tested with the allowed api key and the tests succeeded (great).

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: emily/handler/src/api/handlers/chainstate.rs [hunk 1]
```diff
@@ -171,22 +180,28 @@ pub async fn update_chainstate(
...
+                    canonical_tip: chainstate.clone(),
+                    conflicting_chainstates,
+                };
+                // Execute the reorg.
+                execute_reorg_handler(context, execute_reorg_request)
+                    .await
+                    .inspect_err(|e| warn!("Failed executing reorg with error {}", e))?;
+            // Log error.
+            } else {
+                info!("Inconsistent chainstate found for {entry:?} but we pretend it's okay.");
```
Comment (cylewitruk) [inline emily/handler/src/api/handlers/chainstate.rs:203]: I'd probably change this to like trace or debug since it'll probably happen a lot?

**Review by cylewitruk — APPROVED** [review]
I think this looks okay, just one comment re: logging (so it's easier to filter out this stuff), but no need to fix it.

As mentioned in other channels, I don't think we should use env vars for the trusted api key if this "fix" were to live long enough for a mainnet release, just because env vars have a tendency to "leak".

File: emily/handler/src/api/handlers/chainstate.rs (hunk 1 above)
Comment (AshtonStephens) [reply emily/handler/src/api/handlers/chainstate.rs:203]: Okay

Comment (AshtonStephens) [reply emily/handler/src/api/handlers/chainstate.rs]: Done.

**Pull request merged**